android/app/src/main/java/org/fossasia/openevent/core/main/DownloadHandler.java

```diff
@@ -0,0 +1,152 @@
+package org.fossasia.openevent.core.main;
+
+import android.content.Context;
+
+import org.fossasia.openevent.common.ConstantStrings;
+import org.fossasia.openevent.common.api.APIClient;
+import org.fossasia.openevent.common.api.Urls;
+import org.fossasia.openevent.common.utils.SharedPreferencesUtil;
+import org.fossasia.openevent.data.repository.RealmDataRepository;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.realm.Realm;
+import timber.log.Timber;
+
+public class DownloadHandler {
+
+    private final Context context;
+
+    public DownloadHandler(Context context) {
+        this.context = context;
+    }
+
+    private static final RealmDataRepository realmDataRepository = RealmDataRepository.getInstance(Realm.getDefaultInstance());
+
+    public Observable<String> downloadAllAssets() {
+
+        return Observable.fromArray(ConstantStrings.EVENT,
+                ConstantStrings.SESSIONS,
+                ConstantStrings.SPEAKERS,
+                ConstantStrings.TRACKS,
+                ConstantStrings.SPONSORS,
+                ConstantStrings.MICROLOCATIONS,
+                ConstantStrings.SESSION_TYPES)
+                .flatMapSingle(this::downloadAsset);
+    }
```

Comment:
Rather than this, create isolated functions for downloading each part, like getEvent, getSessions, getSpeakers, etc and then it will be chained by a single method